balloc/balloc.c

```diff
@@ -640,9 +646,13 @@ balloc_group_write_credit(struct m0_balloc               *bal,
                           struct balloc_groups_write_cfg *bgs,
                           struct m0_be_tx_credit         *credit)
 {
-	m0_be_btree_insert_credit(&bal->cb_db_group_extents, 2,
-		M0_MEMBER_SIZE(struct m0_ext, e_start),
-		M0_MEMBER_SIZE(struct m0_ext, e_end), credit);
+	int i;
+
+	for (i = 0; i < bal->group_count; i++) {
+		m0_be_btree_insert_credit(&bal->cb_db_group_extents[i], 2,
+			M0_MEMBER_SIZE(struct m0_ext, e_start),
+			M0_MEMBER_SIZE(struct m0_ext, e_end), credit);
+	}
```

Comment:
Are you sure that we need to update *ALL* btrees in this tx?

Maybe we only need to update modified btree, not *ALL* tree?

Reply:
Following function will be called during MKFS,  we are creating all persistent btrees Node here. Please correct me if i am wrong here, I think we need all nodes credit for complete transaction.

Reply:
@shashank-parulekar , please  correct me if i am wrong here, i think balloc_group_write  uses balloc_group_write_credit  that at time create multiple btrees, without loop, it may fail with insufficient  credit.